src/main/java/com/google/collegeplanner/data/Meeting.java

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package com.google.collegeplanner.data;
+
+package com.google.collegeplanner.data;
+
+import java.text.ParseException;
+import java.time.DayOfWeek;
+import java.util.ArrayList;
+import java.util.Collections;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+public class Meeting {
+  private ArrayList<DayOfWeek> days;
+  private String room;
+  private String building;
+  private int startTime;
+  private int endTime;
+
+  public Meeting(String days, String room, String building, 
+  int startTime, int endTime) throws ParseException {
+    this.days = new ArrayList<DayOfWeek>();
+    this.room = room;
+    this.building = building;
+    this.startTime = startTime;
+    this.endTime = endTime;
+
+    if (days.toUpperCase().contains("M")) {
+      this.days.add(DayOfWeek.MONDAY);
+    }
+    if (days.toUpperCase().contains("TU")) {
+      this.days.add(DayOfWeek.TUESDAY);
+    }
+    if (days.toUpperCase().contains("W")) {
+      this.days.add(DayOfWeek.WEDNESDAY);
+    }
+    if (days.toUpperCase().contains("TH")) {
+      this.days.add(DayofWeek.THURSDAY);
+    }
+    if (days.toUpperCase().contains("F")) {
+      this.days.add(DayOfWeek.FRIDAY);
+    }
+    if (days.toUpperCase().contains("SA") || days.toUpperCase().contains("SU")) {
+      throw new ParseException("Invalid Day, no Weekends on Academic Calendar");
+    }
+  }
+
+  public Meeting(JSONObject json) throws ParseException {
+    this((String) json.get("days"), (String) json.get("room"), (String) json.get("building"),
+        (int) json.get("start_time"), (int) json.get("end_time"));
+  }
+
+  public boolean conflictsWith(Meeting other) {
+    // Checks to see if the two meetings occur on the same day
+    // If not, no need to check the time as they cannot conflict
+    boolean canSkip = true;
+    for (String day : other.getDays()) {
+      if (days.contains(day)) {
+        canSkip = false;
+        break;
+      }
+    }
+
+    if (canSkip) {
+      return false;
+    }
+
+    if (startTime > other.getStartTime()) {
+      if (startTime >= other.getEndTime()) {
+        return false;
+      }
+    } else {
+      if (endTime <= other.getStartTime()) {
+        return false;
+      }
+    }
```

Comment:
The logic here seems wrong.
```suggestion
    if (startTime > other.getStartTime()) {
      if (endTime >= other.getStartTime()) {
        return false;
      }
    } else {
      ...
    }
```
Or you can simply check `endTime >= other.getStartTime || startTime <= other.getEndTime` if each Meeting is a valid one (start time always > end time).

Reply:
In the first block of the if statement, I am checking if the current meeting's startTime is greater than other meeting's startTime. If that is true, then that means that the current meeting starts after the other meeting. So in order not to conflict with each other, the current meeting must start after the other meeting's endTime. There is no reason to check `endTime >= other.getStartTime()` if we already check if endTime is greater than startTime.

Reply:
Oh yes, my bad, I somehow was thinking bigger number time is ahead of smaller number time.